File: benchmarks/tasks/record_transform/record_transform.hs
-- SPDX-License-Identifier: MIT-0
module Main where

data Person = Person { name :: String, x :: Int, y :: Int }

display :: Person -> String
display p = name p ++ " at (" ++ show (x p) ++ ", " ++ show (y p) ++ ")"

main :: IO ()
main = putStrLn (display updated)
  where
    person  = Person { name = "Alice", x = 1, y = 2 }
    updated = person { name = "Bob", x = 3, y = 4 }

File: benchmarks/tasks/record_transform/record_transform.py
# SPDX-License-Identifier: MIT-0

p = {"name": "Alice", "x": 0, "y": 0}
moved = {**p, "x": 3, "y": 4}
renamed = {**moved, "name": "Bob"}
print(f"{renamed['name']} at ({renamed['x']}, {renamed['y']})")
